src/pages/list/table-list/components/StandardTable/index.tsx

```diff
@@ -0,0 +1,164 @@
+import { Alert, Table } from 'antd';
+import { ColumnProps, TableRowSelection, TableProps } from 'antd/es/table';
+import React, { Component, Fragment } from 'react';
+
+import { TableListItem } from '../../data.d';
+import styles from './index.less';
+
+type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;
+
+export interface StandardTableProps<T> extends Omit<TableProps<T>, 'columns'> {
+  columns: StandardTableColumnProps[];
+  data: {
+    list: TableListItem[];
+    pagination: StandardTableProps<TableListItem>['pagination'];
+  };
+  selectedRows: TableListItem[];
+  onSelectRow: (rows: any) => void;
+}
+
+export interface StandardTableColumnProps extends ColumnProps<TableListItem> {
+  needTotal?: boolean;
+  total?: number;
+}
+
+function initTotalList(columns: StandardTableColumnProps[]) {
+  if (!columns) {
+    return [];
+  }
+  const totalList: StandardTableColumnProps[] = [];
+  columns.forEach(column => {
+    if (column.needTotal) {
+      totalList.push({ ...column, total: 0 });
+    }
+  });
+  return totalList;
+}
+
+interface StandardTableState {
+  selectedRowKeys: string[];
+  needTotalList: StandardTableColumnProps[];
+}
+
+class StandardTable extends Component<StandardTableProps<TableListItem>, StandardTableState> {
+  static getDerivedStateFromProps(nextProps: StandardTableProps<TableListItem>) {
+    // clean state
+    if (nextProps.selectedRows.length === 0) {
+      const needTotalList = initTotalList(nextProps.columns);
+      return {
+        selectedRowKeys: [],
+        needTotalList,
+      };
+    }
+    return null;
+  }
+
+  constructor(props: StandardTableProps<TableListItem>) {
+    super(props);
+    const { columns } = props;
+    const needTotalList = initTotalList(columns);
+
+    this.state = {
+      selectedRowKeys: [],
+      needTotalList,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The key 'needTotalList' is not sorted alphabetically](https://app.codacy.com/app/afc163/ant-design-pro/pullRequest?prid=4583897)